File: app/src/main/java/com/scatl/uestcbbs/module/medal/adapter/MedalCenterAdapter.java
package com.scatl.uestcbbs.module.medal.adapter;

import android.view.View;
import android.widget.Button;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.MedalBean;
import com.scatl.uestcbbs.helper.glidehelper.GlideLoader4Common;

/**
 * author: sca_tl
 * date: 2020/10/7 19:06
 * description:
 */
public class MedalCenterAdapter extends BaseQuickAdapter<MedalBean.MedalCenterBean, BaseViewHolder> {
    public MedalCenterAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, MedalBean.MedalCenterBean item) {
        helper.setText(R.id.item_model_center_name, item.medalName)
                .setText(R.id.item_medal_center_dsp, item.medalDsp)
                .addOnClickListener(R.id.item_model_center_get_medal_btn);

        Button button = helper.getView(R.id.item_model_center_get_medal_btn);
        if (item.buyDsp == null || item.buyDsp.length() == 0) {
            button.setVisibility(View.GONE);
        } else {
            button.setVisibility(View.VISIBLE);
            button.setText(item.buyDsp);
        }

        GlideLoader4Common.simpleLoad(mContext, item.medalIcon, helper.getView(R.id.item_model_center_icon));
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/medal/adapter/MedalCenterAdapter.kt
package com.scatl.uestcbbs.module.medal.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemMedalCenterBinding
import com.scatl.uestcbbs.entity.MedalBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/12 15:07
 */
class MedalCenterAdapter: PreloadAdapter<MedalBean.MedalCenterBean, ItemMedalCenterBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemMedalCenterBinding {
        return ItemMedalCenterBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemMedalCenterBinding>, position: Int, item: MedalBean.MedalCenterBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.medalName.text = item.medalName
        holder.binding.medalDsp.text = item.medalDsp
        holder.binding.medalIcon.load(item.medalIcon)

        if (item.buyDsp.isNullOrEmpty()) {
            holder.binding.getMedalBtn.visibility = View.GONE
        } else {
            holder.binding.getMedalBtn.visibility = View.VISIBLE
            holder.binding.getMedalBtn.text = item.buyDsp
        }
    }
}